CIP-0060/cddl/version-1.cddl

```diff
@@ -0,0 +1,47 @@
+string = text .size (0..64)
+
+policy_id = string / bytes ; hex string for CIP-25 version 1, bytes version 2
+asset_name = string / bytes ; utf-8 for CIP-25 version 1, bytes for version 2
+
+artist_details =
+  {
+    name : string,                                                             ; artist or band name
+    ? image : string / [* string],      ; optional image of the artist
+  }
+
+files_details = 
+  {
+    name : string,
+    mediaType : string,
+    src : string / [* string],
+    ? artists : [* artist_details],
+    ? album_title : string,
+    ? track_number: uint,
+    ? song_title: string / [* string],
+    ? song_duration: string .regexp "^P(?!$)(T(?=\d)(\d+H)?(\d+M)?(\d+S)?)$",    ; iso8601 duration
+    ? genres: [1*3 string],
+    ? copyright: string,
+  }
+
+metadata_details = 
+  {
+    name : string,
+    image : string / [* string],
+    artists : [* artist_details],
```

Comment:
@SebastienGllmt Does CDDL have the concept of something being optional based on a condition? For example, if `"release_type": "Multiple"`, then many of the top level fields should be optional, because they'll be defined many times at the lower levels. 

Or do I just have to make everything optional and explain it with comments?

Reply:
Alternatively, should I just define two separate .cddl specs with one for "Single" and one for "Multiple"?